src/VisualStudio/Core/Def/Implementation/Preview/FileChange.cs

```diff
@@ -225,14 +231,51 @@ internal override void GetDisplayData(VSTREEDISPLAYDATA[] pData)
             pData[0].Image = pData[0].SelectedImage = (ushort)StandardGlyphGroup.GlyphLibrary;
         }
 
-        public static IHierarchicalDifferenceCollection ComputeDiffSpans(SourceText oldText, SourceText newText, ITextDifferencingService diffService, CancellationToken cancellationToken)
+        private static IHierarchicalDifferenceCollection ComputeDiffSpans(ITextDifferencingService diffService, TextDocument left, TextDocument right, CancellationToken cancellationToken)
         {
-            var diffResult = diffService.DiffStrings(oldText.ToString(), newText.ToString(), new StringDifferenceOptions()
+            // TODO: it would be nice to have a syntax based differ for presentation here, 
+            //       current way of just using text differ has its own issue, and using syntax differ in compiler that are for incremental parser
+            //       has its own drawbacks.
+
+            var oldText = left.GetTextAsync(cancellationToken).WaitAndGetResult(cancellationToken);
+            var newText = right.GetTextAsync(cancellationToken).WaitAndGetResult(cancellationToken);
+
+            var oldString = oldText.ToString();
+            var newString = newText.ToString();
+
+            // first try, cheapest way.
+            var diffResult = diffService.DiffStrings(oldString, newString, new StringDifferenceOptions()
             {
-                DifferenceType = StringDifferenceTypes.Line
+                DifferenceType = StringDifferenceTypes.Line | StringDifferenceTypes.Word,
```

Comment:
let me finalize the diffing work I started. please put the link - https://github.com/dotnet/roslyn/pull/2098 to the new bug. so that I can track these issue as well.